src/main/java/hudson/plugins/jacoco/JacocoBuildAction.java

```diff
@@ -33,9 +36,9 @@
  * @author Jonathan Fuerth
  * @author Ognjen Bubalo
  */
-public final class JacocoBuildAction extends CoverageObject<JacocoBuildAction> implements HealthReportingAction, StaplerProxy, Serializable {
+public final class JacocoBuildAction extends CoverageObject<JacocoBuildAction> implements HealthReportingAction, StaplerProxy, Serializable, RunAction2 {
 
-	public final AbstractBuild<?,?> owner;
+	public transient Run<?,?> owner;
```

Comment:
The problem is that existing (already compiled) code may not be able to deal with the type change. I'm not sure in the case of a field though @jtnord , do you know for sure?

BCT is the [bytecode compatibility transformer](https://github.com/jenkinsci/bytecode-compatibility-transformer).